protobuf-codegen/src/main/proto/messages.proto

```diff
@@ -0,0 +1,356 @@
+/*
+ * Wire
+ * Copyright (C) 2021 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+syntax = "proto2";
+
+import "qualified_user_id.proto";
+
+message GenericMessage {
```

Comment:
The proto files are copied into the project now? This makes it hard to know which of the protobuf file we are using in Kalium. 

Can we pull them GitHub only define the release tag in the project. If this is not easy we could do it in a separate PR  but I think we should do it.

Reply:
Not 100% copy pasted. As I mentioned in the PR description, the QualifiedUserID, for example, needed to be extracted to a common file.

Some of the Java options supported by Protoc (like outer class name or adding a suffix) are not respected when using PBandK.

I'd like to streamline the process and checkout from the otr-proto repo, but it's far from easy to do so right now. Maybe it would help by extracting the QualifiedUserID in the original otr-proto repo, but that would be a breaking change.

I don't have any solutions now.

Reply:
The `otr.proto` and `messages.proto` are supposed to live in different namesakes doesn't PBandK respect that? See https://github.com/wireapp/generic-message-proto/pull/74#pullrequestreview-680786841

Reply:
So yeah, seems like https://github.com/streem/pbandk/issues/179 doesn't respect `package` but also the `option java_package` is different so I don't understand why the clash.

Reply:
Right so it clashes because package name inside the proto files are override by the plugin option:
`option("kotlin_package=com.wire.kalium.protobuf")`

Reply:
TBH I felt like PBandK doesn't respect a lot of these options yet. I tried playing with them before submitting the PR and in the end I just removed them, as they didn't have any effect on the generated code.
I didn't want to leave `package` and `option`s in there, so they would suddenly kick-in when we bump PBandK's version or someone spends time trying to understand these lines of code in the future.

It's a trade-off for now. We increase a bit of maintenance with `.proto` definitions, but as a pro: we have a multi-platform solution. In my opinion, it's worth it. But I'd like a second confirmation before we move on.

Hopefuly these will be supported Soon™.

Reply:
Yes, it's definitely worth it! Just looking for way to avoid having to copy & paste the files and edit them. But like I said earlier I'm not pushing for solving it in this PR.

Reply:
> Right so it clashes because package name inside the proto files are override by the plugin option: `option("kotlin_package=com.wire.kalium.protobuf")`

Indeed, the `kotlin_package` was overwriting it 🤦🏼‍♂️. Removing that made the `package` work.

Still, the original `messages.proto` doesn't have a `package`. So we still need to modify the files a tiny bit when getting from the original repo. But the changes are one-liner.